### ISSUE TITLE
Strange error when setting a wrong username

### ISSUE THREAD

**Issue by josemazo** [CartoDB/cartoframes#1466]
Really this is a problem that comes from `carto-python`, or even from `pyrestcli`, but it returns a confusing error, even for experienced developers. So maybe, we should forget it, but let's document it.

If you set a wrong username in the credentials, the returned error is a `NotFoundException` with a huge HTML document. 

```python
from cartoframes import read_carto, to_carto
from cartoframes.auth import set_default_credentials

set_default_credentials(
    username='josema_not_found_account',
    api_key='josema_not_found_account_api_key'
)

gdf = read_carto('lanius_collurio')
```

The realised exceptions message starts with:
```
NotFoundException: <!DOCTYPE html> <html> <meta charset=utf-8> ...
```

**Comment by Jesus89** [comment]
Good catch. Maybe we can filter the exception `NotFoundException` form CF when using `carto-python` client calls and raise a better exception.

**Comment by Jesus89** [comment]
However, this should be fixed in `carto-python` or `pyrescli` because otherwise, the Traceback will contain the 404 HTML page.

**Comment by Jesus89** [comment]
https://app.clubhouse.io/cartoteam/story/93710/fix-wrong-username-error

**Comment by Jesus89** [comment]
Closed in #1687
